sanity-io/sdk
# feat: add session hooks and store

### Description

This PR adds a zustand store and react hooks for the auth logic and [sessionState](https://sdk-docs-git-feat-add-session-hooks-and-store.sanity.dev/interfaces/_sanity_sdk.SessionState.html). It adds
* `SessionStore` 
* `useLoggedInState`
* `useCurrentUser`

![CleanShot 2024-12-02 at 16 07 18@2x](https://github.com/user-attachments/assets/b6b9c5a3-a089-4e43-ba87-1594ad5cef1f)




### What to review
Review the items listed above and the way that hooks are created via Zustand's `useStore` function.

### Testing
1. Run this locally
2. Log in
3. Verify that you are logged in
4. Check the Redux devtools for the actions and state changes

### Fun gif
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNjJmbDNtOG42emxqZXhxeGJocjJ0bmwwMHo2MXlmYXhjM2dpY2U4eiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ch5VA7b4Qq8XCddVlN/giphy-downsized.gif)
